### PR TITLE
Harden import validation and CLI parsing

### PR DESCRIPTION
**Pull request by esnya**
## Summary
- normalize import requests before validation and validate remote archive URLs plus package-keyed maps consistently
- accept package alias pattern flags, tighten CLI remote URL parsing, and add PlateauMeshCode coverage for invalid subdivisions
- make live asset state writes safe for shared work roots and fix the remote README examples in English and Japanese

## Verification
- `dotnet test Plateau.ResoniteLink.sln --configuration Release -m:1 -p:UseSharedCompilation=false`
- `dotnet test tests/Plateau.ResoniteLink.Tests/Plateau.ResoniteLink.Tests.csproj --configuration Release --filter "FullyQualifiedName~PlateauImportRequestValidatorTests|FullyQualifiedName~PlateauImportServiceTests|FullyQualifiedName~CliArgumentsParserTests|FullyQualifiedName~PlateauMeshCodeTests|FullyQualifiedName~ResoniteLinkSceneBuilderAssetReuseTests" -m:1 -p:UseSharedCompilation=false`

## Notes
- `dotnet format whitespace . --folder --verify-no-changes` timed out after 60s in this WSL environment, so whitespace verification could not be completed with that command.

### CODE REVIEW

File: src/Plateau.ResoniteLink.Application/Importing/PlateauImportService.cs [hunk 1]
```diff
@@ -24,22 +24,13 @@ public async Task<ImportExecutionResult> ExecuteAsync(
     {
         ArgumentException.ThrowIfNullOrWhiteSpace(workRoot);
 
-        IReadOnlyList<string> validationErrors = PlateauImportRequestValidator.Validate(request);
+        PlateauImportRequest normalizedRequest = NormalizeRequest(request);
+        IReadOnlyList<string> validationErrors = PlateauImportRequestValidator.Validate(normalizedRequest);
```
[the remark this answers is not in the document]
Comment (esnya) [reply src/Plateau.ResoniteLink.Application/Importing/PlateauImportService.cs]: Fixed the exception path on the follow-up branch. `PlateauImportService` now does a trim-only normalization before validation, so unsupported package names/keys are still reported through `PlateauImportValidationException`, and canonical package-map normalization happens only after validation passes. I also added a regression test covering an unsupported `PackagePatterns` key.

[the remark this answers is not in the document]
Comment (esnya) [reply src/Plateau.ResoniteLink.Application/Importing/PlateauImportService.cs]: Handled in the follow-up branch. `PlateauImportService` now does a trim-only normalization pass before validation, validates that request, and only canonicalizes package names/maps after validation succeeds. Unsupported package names therefore still surface as `PlateauImportValidationException` with structured errors instead of leaking `ArgumentException`. I also added a regression test for an unsupported package-pattern key.

**Comment by esnya** [comment]
@codex review

**Comment by esnya** [comment]
@codex review

README / help text の整合修正を追加しました。差分の確認をお願いします。

File: src/Plateau.ResoniteLink.Application/Importing/PlateauImportService.cs [hunk 2]
```diff
@@ -99,4 +92,87 @@ private void ReportProgress(string message)
...
     }
+
+    private static PlateauImportRequest NormalizeRequestForValidation(PlateauImportRequest request)
+    {
+        ArgumentNullException.ThrowIfNull(request);
+
+        return request with
+        {
+            Dataset = request.Dataset.Trim(),
+            MeshCode = request.MeshCode.Trim(),
```
[the remark this answers is not in the document]
Comment (esnya) [reply src/Plateau.ResoniteLink.Application/Importing/PlateauImportService.cs]: Handled in `e4acb0b`. `NormalizeRequestForValidation` no longer calls `.Trim()` directly on required string fields; it now uses a null-safe trim-to-empty pass so deserialized or partially populated requests still flow into `PlateauImportRequestValidator` and surface as `PlateauImportValidationException`. I also added a regression test covering null `Dataset` and `MeshCode` inputs.

File: src/Plateau.ResoniteLink.Application/Importing/PlateauImportService.cs [hunk 3]
```diff
@@ -99,4 +92,87 @@ private void ReportProgress(string message)
...
+            LocalSourcePath = string.IsNullOrWhiteSpace(request.LocalSourcePath) ? null : request.LocalSourcePath.Trim(),
+            PackageNames = request.PackageNames is null
+                ? null
+                : request.PackageNames.Select(static packageName => packageName.Trim()).ToArray(),
+            ExcludeLodLevelsByPackage = request.ExcludeLodLevelsByPackage is null
+                ? null
+                : request.ExcludeLodLevelsByPackage.ToDictionary(
+                    static pair => pair.Key.Trim(),
+                    static pair => pair.Value,
+                    StringComparer.OrdinalIgnoreCase),
```
[the remark this answers is not in the document]
Comment (esnya) [reply src/Plateau.ResoniteLink.Application/Importing/PlateauImportService.cs]: Handled in `e4acb0b`. Pre-validation no longer rebuilds package maps, so trim/case collisions do not throw before validation. Instead, `PlateauImportRequestValidator` now trims package keys when validating, detects duplicate normalized package keys for both `ExcludeLodLevelsByPackage` and `PackagePatterns` (including alias collisions like `waterbody` / `wtr`), and reports them through `PlateauImportValidationException`. Regression tests were added for both service and validator paths.

**Comment by esnya** [comment]
@codex review

review コメント対応と追加の一巡を反映しました。`e4acb0b` で pre-validation の null-safe 化と package-map 正規化衝突の validation 化を入れ、`dotnet test Plateau.ResoniteLink.sln --configuration Release -m:1 -p:UseSharedCompilation=false` と `dotnet format whitespace . --folder --verify-no-changes` は通過しています。差分の再確認をお願いします。

File: src/Plateau.ResoniteLink.Application/Importing/PlateauImportService.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (esnya) [reply src/Plateau.ResoniteLink.Application/Importing/PlateauImportService.cs]: Handled in `e4acb0b`. Pre-validation no longer normalizes package maps into case-insensitive dictionaries, so trim/case collisions cannot throw `ArgumentException` before validation. The validator now detects duplicate normalized package keys and reports them as structured validation errors instead; I also added regression coverage for both whitespace/case collisions and alias collisions.

File: src/Plateau.ResoniteLink.Cli/ResoniteLinkSceneBuilder.cs [hunk 4]
```diff
@@ -1716,7 +1717,9 @@ private async Task PersistAssetSourceFingerprintsAsync(CancellationToken cancell
         {
             string directory = Path.GetDirectoryName(liveAssetStatePath) ?? ".";
             Directory.CreateDirectory(directory);
-            string temporaryPath = $"{liveAssetStatePath}.tmp";
+            string temporaryPath = Path.Combine(
+                directory,
+                $"{Path.GetFileName(liveAssetStatePath)}.{Guid.NewGuid():N}.tmp");
```
[the remark this answers is not in the document]
Comment (esnya) [reply src/Plateau.ResoniteLink.Cli/ResoniteLinkSceneBuilder.cs:1722]: Handled in `2b7a703`. `PersistAssetSourceFingerprintsAsync` now wraps each GUID-scoped temp file in a `try/finally` and does best-effort deletion if `WriteAllTextAsync` or `File.Move` fails, so interrupted or rejected writes no longer leak `resonite-live-asset-state.json.*.tmp` files into the shared work root. I also added a regression test that forces the final move to fail by pre-creating `resonite-live-asset-state.json` as a directory and verifies no temp files are left behind.

**Comment by esnya** [comment]
@codex review

追加の review コメントを `2b7a703` で対応しました。live asset state の GUID temp file を失敗時に best-effort cleanup するようにし、`File.Move` 失敗を再現する回帰テストも追加しています。`dotnet test Plateau.ResoniteLink.sln --configuration Release -m:1 -p:UseSharedCompilation=false` は `155/155`、`dotnet format whitespace . --folder --verify-no-changes` も通過しています。差分の再確認をお願いします。
